otelcollector/deploy/addon-chart/azure-monitor-metrics-addon/templates/_ama-metrics-helpers.tpl

```diff
@@ -0,0 +1,39 @@
+{{/* HPA merge. */}}
+{{/* 
+  1. Set the default HPA values for minReplicas, maxReplicas, and metrics. 
+  2. If the current HPA already exists, override the default HPA values to the current values.
+*/}}
+{{ define "ama-metrics-merge-custom-hpa" }}
+
+{{/* Set the default HPA values for minReplicas, maxReplicas, and metrics.  */}}
+{{- $amaMetricsHpaName := "ama-metrics-hpa" }}
+{{- $amaMetricsAutoscaleMin := 2 -}}
+{{- $amaMetricsAutoscaleMax := 8 -}}
+
+amaMetricsMinReplicasFromHelper: 2
+amaMetricsMaxReplicasFromHelper: 8
+
+{{/* If the current HPA already exists, set the HPA values to the current 
+     HPA spec to preserve those values. */}}
+
+{{- $amaMetricsCurrentHPA := lookup "autoscaling/v2" "HorizontalPodAutoscaler" "kube-system" $amaMetricsHpaName }}
+{{- if and $amaMetricsCurrentHPA $amaMetricsCurrentHPA.spec }}
+{{- $amaMetricsMinReplicasFromCurrentSpec := $amaMetricsCurrentHPA.spec.minReplicas -}}
+{{- $amaMetricsMaxReplicasFromCurrentSpec := $amaMetricsCurrentHPA.spec.maxReplicas -}}
+
+  {{- if and ($amaMetricsMinReplicasFromCurrentSpec) (gt (int $amaMetricsMinReplicasFromCurrentSpec) 0) }}
```

Comment:
Not sure why we are comparing min from current spec , again with min from current spec ?

Reply:
The first check(line 24) is for checking if it is greater than 0 and the second(line 25) is to make sure it is greater than the minimum we want to support which is 2. 
Pls lmk if I got the lines wrong and explaining the wrong checks :)